students/morgan/session03/mailroom2.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+# name, donation total, number of donations and average of donations
+
+infile = 'donors.txt'
+
+donors = {}
+
+with open(infile) as donor_input:
+    donor_input.readline()
+    for line in donor_input:
+        # line = line.strip()
+        name_string = line.split('-')[0].strip()
+        donation_string = line.split('-')[1].strip(' ').strip()
+        donation_list = donation_string.split(',')
+        
+        donation_list = [float(x) for x in donation_list]
```

Comment:
nice! aren't comprehensions good for this sort of thing?

and float() will ignore whitespace, so OK if you didn't call strip() everywhere.